### DISCUSSION TITLE
Why can't my code modify the metadata of images in the current folder using Python to call ImageMagick?

### DISCUSSION THREAD

**Discussion by asairhwb** [ImageMagick/ImageMagick#7645] [Help] [answered]
# Problem description:
Why can't my code modify the metadata of images in the current folder using Python to call ImageMagick?
The code runs without errors, but the metadata of the images doesn't seem to change at all, and I can't find the reason. 
I am a beginner. Please help me，thank you!
# Version of the tools:
ImageMagick version
Version: ImageMagick 7.1.1-6 Q16-HDRI x64 b2dd67b:20230402 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0) 
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193431943)

Windows System: Windows 11
Windows Version: 10.0.22631

Python : 3.11.3 (tags/v3.11.3:f3909b8, Apr  4 2023, 23:49:59) [MSC v.1934 64 bit (AMD64)]

# Python code
import os
import subprocess
from concurrent.futures import ThreadPoolExecutor, as_completed
from tqdm import tqdm
from colorama import init

init(autoreset=True)
image_extensions = ('.jpg', '.jpeg', '.png', '.webp')
quality = '100'
text = "'Rose is a rose'"
max_workers = 10  # Set maximum number of worker threads

def remove_metadata(file_path, remove_all):
    """Use magick mogrify command to strip metadata from image files"""
    try:
        if remove_all:
            subprocess.run(['magick', 'mogrify', '-strip', file_path], check=True)
        else:
            metadata_fields = [
                'artist', 'Caption', 'Comment', 'label', 'title', 'author', 'exif:artist', 
                'exif:Copyright', 'exif:ImageDescription', 'WinXP-Author', 'WinXP-Comments', 
                'WinXP-Keywords', 'WinXP-Subject', 'WinXP-Title', 'MicrosoftPhoto:CameraSerialNumber', 
                'MicrosoftPhoto:FlashManufacturer', 'MicrosoftPhoto:FlashModel', 'MicrosoftPhoto:LensManufacturer', 
                'MicrosoftPhoto:LensModel', 'xmp:Rating'
            ]
            command = ['magick', 'mogrify', file_path, "-quality", quality] + [item for field in metadata_fields for item in ['-set', field, text]]
            subprocess.run(command, check=True)
        return f"Successfully stripped metadata: {file_path}"
    except subprocess.CalledProcessError as e:
        return f"Error: An error occurred while processing file '{file_path}'. Error message: {e}"

def main():
    current_directory = os.getcwd()
    files = [os.path.join(current_directory, filename) for filename in os.listdir(current_directory) if filename.lower().endswith(image_extensions)]

    if not files:
        print("Error: No image files found in the current directory.")
        return

    view_metadata = input("Do you want to view the metadata? (Press Enter to skip, type Y or yes to view metadata): ").strip().lower()
    if view_metadata in ['y', 'yes']:
        for file in files:
            print(f"Metadata for {file}:")
            subprocess.run(['magick', 'identify', '-verbose', file])

    print(f"Text: {text}")
    remove_option = input("Please enter the removal method (Press Enter to default to removing only tags, titles, authors, ratings, and subjects; type A or ALL to remove all metadata): ").strip().upper()
    remove_all = remove_option in ['A', 'ALL']

    with ThreadPoolExecutor(max_workers=max_workers) as executor:
        with tqdm(total=len(files), desc="Stripping metadata", unit="file") as pbar:
            futures = [executor.submit(remove_metadata, file, remove_all) for file in files]
            for future in as_completed(futures):
                result = future.result()
                print(result)
                pbar.update(1)

    restart = input("Do you want to restart? Type 'R' to rerun, press Enter to stop the program: ").strip().upper()
    if restart == "R":
        main()
    else:
        print("Program ended.")
        input("Press any key to close the program...")

if __name__ == "__main__":
    main()
# Test image
![Lee-Ahrin-Likey-Web-Pictorial-hotgirl world-041](https://github.com/user-attachments/assets/07e0cbc6-5bcc-4bb7-89b5-62abdb238545)

**Comment by fmw42** [comment, marked as answer]
Imagemagick does not allow you to modify meta data in general. Use Exiftool, if you want to do that. 

In Imagemagick you can modify the comment and the label using -set comment "Your Comment" and -set label "Your Label".  Other meta fields and the Exif fields are not modifiable.

**Reply by asairhwb** [reply]
I see.Thank you.
